src/comotion/dash.py

```diff
@@ -567,46 +860,72 @@ def read_and_upload_file_to_dash(
     List
         List of http responses
     """
-    file_reader = pd.read_csv(
-        file,
-        chunksize=chunksize,
-        encoding=encoding,
-        dtype=str  # Set all columns to strings.  Dash will still infer the type, but this makes sure it doesnt mess with the contents of the csv before upload
-    )
-
-    i = 1
-    responses = []
-    for file_df in file_reader:
-
-        if modify_lambda is not None:
-            modify_lambda(file_df)
-
-
-        csv_stream = create_gzipped_csv_stream_from_df(file_df)
-
-        if path_to_output_for_dryrun is None:
-
-            response = upload_csv_to_dash(
-                dash_orgname=dash_orgname,
-                dash_api_key=dash_api_key,
-                dash_table=dash_table,
-                csv_gz_stream=csv_stream,
-                service_client_id=service_client_id
-            )
-
-            responses.append(response.text)
+    
+    if not data_model_version or data_model_version not in ['v1', 'v2']:
+        print("Determining Data Model Version")
+        try:
+            if not application_config:
+                config = DashConfig(Auth(orgname=dash_orgname))
+            else:
+                if 'client_id' not in application_config or 'secret' not in application_config:
+                    raise ValueError("Please provide these keys in the application_config: ['client_id', 'secret']")
+                else:
+                    config = DashConfig(Auth(entity_type=Auth.APPLICATION,
+                                            application_client_id=application_config['client_id'],
+                                            application_client_secret=application_config['secret'],
+                                            orgname=dash_orgname))
+
+            # Get migration status
+            migration = Migration(config)
+            print('Migration Status: ' + str(migration.status().full_migration_status))
+            if migration.status().full_migration_status == 'Completed': # What about for new clients who start on v2 lake?  Will status be complete?
```

Comment:
Good question. Until further notice, we must run a migration on all new clients as part of setting them up.